packages/ui-server/utils/user.js

```diff
@@ -1,37 +1,57 @@
 "use strict";
+// Async - Await polyfill
+var __awaiter = (this && this.__awaiter) || function(thisArg, _arguments, P, generator) {
+	return new (P || (P = Promise))(function(resolve, reject) {
+		function fulfilled(value) {
+			try {
+				step(generator.next(value));
+			} catch (e) {
+				reject(e);
+			}
+		}
+		function rejected(value) {
+			try {
+				step(generator["throw"](value));
+			} catch (e) {
+				reject(e);
+			}
+		}
+		function step(result) {
+			result.done ? resolve(result.value) : new P(function(resolve) {
+				resolve(result.value);
+			}).then(fulfilled, rejected);
+		}
+		step((generator = generator.apply(thisArg, _arguments || [])).next());
+	});
+};
 Object.defineProperty(exports, "__esModule", { value: true });
 const net = require("net");
+const p_event_1 = require("p-event");
 class User {
-    constructor(port, address) {
-        this.socket = new net.Socket();
-        this.port = port;
-        this.address = address;
-        this.init();
-    }
-    answer(ans) {
-        const user = this.socket;
-        return new Promise((resolve, reject) => {
-            user.write(ans);
-            if (ans.action === "exit") {
-                user.destroy();
-            }
-            user.on("data", (data) => {
-                if (data.toJSON().data[0].action === "question") {
-                    resolve(data);
-                }
-                if (data.toJSON().data[0].action === "exit") {
-                    user.destroy();
-                }
-            });
-            user.on("error", (err) => {
-                reject(err);
-                user.destroy();
-            });
-        });
-    }
-    init() {
-        const user = this.socket;
-        user.connect(this.port, this.address);
-    }
+	constructor(port, address) {
+		this.socket = new net.Socket();
+		this.port = port;
+		this.address = address;
+		this.init();
+	}
+	answer(ans) {
+		const user = this.socket;
+		user.write(JSON.stringify(ans));
+	}
+	question() {
```

Comment:
Needed a help here,
How do I make a wait for an event to occur return the data received for callback
`p-event` doesn't seems to work now.